base/file_name_utils.hpp

```diff
@@ -17,7 +17,7 @@ std::string GetFileExtension(std::string const & name);
 /// Get file name from full path.
 void GetNameFromFullPath(std::string & name);
 
-std::string FileNameFromFullPath(std::string path);
+std::string_view FileNameFromFullPath(std::string_view path);
```

Comment:
This pursuit for premature optimizations can make an unexpected behavior in a very simple case like:
s = FileNameFromFullPath(a+b).

I have doubts about this change and don't like the way where this code goes ..